internal/k8s/watch.go

```diff
@@ -48,39 +52,61 @@ func (kCli K8sClient) makeWatcher(f watcherFactory, ls labels.Selector) (watch.I
 }
 
 func (kCli K8sClient) WatchPods(ctx context.Context, ls labels.Selector) (<-chan *v1.Pod, error) {
-	ch := make(chan *v1.Pod)
-	watcher, err := kCli.makeWatcher(func(ns string) watcher {
+	// HACK(dmiller): There's no way to get errors out of an informer. See https://github.com/kubernetes/client-go/issues/155
+	// In the meantime, at least to get authorization and some other errors let's try to set up a watcher and then just
+	// throw it away.
+	_, err := kCli.makeWatcher(func(ns string) watcher {
 		return kCli.core.Pods(ns)
 	}, ls)
 	if err != nil {
-		return nil, errors.Wrap(err, "Pods.Watch")
+		return nil, errors.Wrap(err, "pods.Watch")
 	}
 
-	go func() {
-		for {
-			select {
-			case event, ok := <-watcher.ResultChan():
-				if !ok {
-					close(ch)
-					return
-				}
+	// HACK(dmiller): this can log to stderr. To prevent that we remove the error handler.
```

Comment:
are you sure? i'm pretty sure this logs to klog by default, not stderr, which we already have special handling for, so it should be ok